### DISCUSSION TITLE
How to pass ref to DeckGLOverlay custom component using react-map-gl 7.x & deck.gl?

### DISCUSSION THREAD

**Discussion by kboul** [visgl/deck.gl#7778] [Q&A]
I am migrating my project to react-map-gl 7.x. I was using deck.gl 8.4.6 until now. 
There are several breaking changes and the only source for the new migration is https://deck.gl/docs/api-reference/mapbox/mapbox-overlay#using-with-react-map-gl.
However, I would like to use a ref which currently is not possible using this custom component as it returns null
```
export default function DeckGLOverlay(props) {
  const overlay = useControl(() => new MapboxOverlay(props));
  overlay.setProps(props);
  return null;
}
```

I want to derive the **deckRef** to leverage methods such as `pickMultipleObjects`, `getViewports` etc as I used to do so far. 
Does anyone know how can I do that, adding a ref to the **DeckGLOverlay** component which now is wrapped by Map and is not the parent of Map anymore?

```
<Map
          {...viewport}
          cursor={cursor}
          mapStyle={mapStyle(mapTheme)}
          mapboxAccessToken={process.env.REACT_APP_MAPBOX_TOKEN}
          onMouseMove={handleMouseMove}
          onMove={handleMove}
          ref={mapRef}
        >
          <DeckGLOverlay
            // ref={deckRef}
            controller
            layers={getLayers()}
            // getCursor={getCursor}
            // onHover={handleHover}
          />
          <NavigationControl />
          <FullscreenControl />
...
</Map>
```

**Comment by Pessimistress** [comment]
[useImperativeHandle](https://react.dev/reference/react/useImperativeHandle)
